crates/pathfinder/src/state/merkle_tree.rs

```diff
@@ -437,6 +483,41 @@ impl<T: NodeStorage> MerkleTree<T> {
         Ok(result)
     }
 
+    /// Generates a merkle-proof a given `key`.
+    ///
+    /// Returns vector of [`ProofNode`] which form a chain from the root to the key,
+    /// if it exists, or upto the node which proves that the key does not exist.
```

Comment:
I assume the visualization is like this: root at the top, leaf at the bottom.

```suggestion
    /// if it exists, or down to the node which proves that the key does not exist.
```

Reply:
What sort of strange trees do you have where the roots are in sky and leaves in the earth 🤔 😁

Reply:
True, nevertheless anytime I recall a tree drawing from a cs related course at the uni, literally all of them were upside down :smile:

Reply:
true